src/compile/selection/multi.ts

```diff
@@ -8,21 +8,32 @@ const multi:SelectionCompiler = {
     const proj = selCmpt.project,
         datum  = '(item().isVoronoi ? datum.datum : datum)',
         encodings = proj.map((p) => stringValue(p.encoding)).join(', '),
+        bins = {},
         fields = proj.map((p) => stringValue(p.field)).join(', '),
-        values = proj.map((p) => `${datum}[${stringValue(p.field)}]`).join(', ');
+        values = proj.map((p) => {
+          const c = p.encoding, f = p.field;
+          // Binned fields should capture extents, for a range test against the raw field.
+          return model.fieldDef(c).bin ? (bins[f] = 1,
```

Comment:
Here you use f but not in ll 19. I think in this case it's better not to create an alias.